components/Containers/Sidebar/index.tsx

```diff
@@ -1,26 +1,49 @@
+import { useTranslations } from 'next-intl';
 import type { ComponentProps, FC } from 'react';
 
 import SidebarGroup from '@/components/Containers/Sidebar/SidebarGroup';
-import WithSidebarSelect from '@/components/withSidebarSelect';
+import WithRouterSelect from '@/components/withRouterSelect';
+import { useClientContext } from '@/hooks/react-server';
 
 import styles from './index.module.css';
 
 type SidebarProps = {
   groups: Array<ComponentProps<typeof SidebarGroup>>;
 };
 
-const SideBar: FC<SidebarProps> = ({ groups }) => (
-  <aside className={styles.wrapper}>
-    {groups.map(({ groupName, items }) => (
-      <SidebarGroup
-        key={groupName.toString()}
-        groupName={groupName}
-        items={items}
-      />
-    ))}
-
-    <WithSidebarSelect groups={groups} />
-  </aside>
-);
+const SideBar: FC<SidebarProps> = ({ groups }) => {
+  const t = useTranslations();
+  const { pathname } = useClientContext();
+
+  const selectItems = groups.map(({ items, groupName }) => ({
+    label: groupName,
+    items: items.map(({ label, link }) => ({ value: link, label })),
+  }));
+
+  const currentItem = selectItems
+    .map(item => item.items)
+    .flat()
+    .find(item => pathname === item.value);
+
+  return (
+    <aside className={styles.wrapper}>
+      {groups.map(({ groupName, items }) => (
+        <SidebarGroup
+          key={groupName.toString()}
```

Comment:
```suggestion
          key={groupName}
```

Reply:
.toString() needs to be called, as the type of `groupName` is `FormattedMessage` which isn't necessarily a string.